File: src/abstractFactory/java/factory/ChampionConfig.java
package abstractFactory.java.factory;

/**
 * Created by kvirus on 2019/6/7 19:08
 * Email @ caoyingkui@pku.edu.cn
 * <p>
 * |   *******    **      **     **     **
 * |  **            **  **       **  **
 * |  **              **         ***
 * |  **              **         **  **
 * |   *******        **         **     **
 */
public abstract class ChampionConfig {
    public int health;
    public int attack;
    public int armor;
    public int attackSpeed;
    public int magicResist;
    public int moveSpeed;

    public abstract String getDescription();
}

File: src/abstractFactory/kotlin/factory/ChampionConfig.kt
package abstractFactory.kotlin.factory

/**
 * Created by kvirus on 2019/6/7 20:06
 * Email @ caoyingkui@pku.edu.cn
 *
 * |   *******    **      **     **     **
 * |  **            **  **       **  **
 * |  **              **         ***
 * |  **              **         **  **
 * |   *******        **         **     **
 */
open class ChampionConfig {
    open var health: Int = 1
    open var attack: Int = 1
    open var armor: Int = 1
    open var attackSpeed: Int = 1
    open var magicResist: Int = 1
    open var moveSpeed: Int = 1

    open fun getDescription(): String {
        return ""
    }
}